### ISSUE TITLE
Transform Scaling applied wrong to Children

### ISSUE THREAD

**Issue by schmidt-florian** [FlaxEngine/FlaxEngine#1796]
**Issue description:**
The scaling to the children is applied wrong. Maybe related to: #1350

https://github.com/FlaxEngine/FlaxEngine/assets/14238126/8dc9bf89-c609-41b1-a250-4b4a1494754e

**Steps to reproduce:**
Open attached project.
Open `Content/Scene`
Change scale from Floor on X from 20 to like 50

**Minimal reproduction project:**
[BugTest1.zip](https://github.com/FlaxEngine/FlaxEngine/files/13125217/BugTest1.zip)

**Flax version:**
master 806590d

**Comment by mafiesto4** [comment]
When using:
```cpp
// Transform::LocalToWorld
(GetWorld() * other.GetWorld()).Decompose(result);
```
Fixes the scaling but if there are additional transformations (eg. that whole object has another parent with some rotation) then we end up with weird issues:

https://github.com/FlaxEngine/FlaxEngine/assets/3464853/a555c533-32fc-4f12-93d0-581ac9489214

We need to find a better solution to this problem.

**Comment by mafiesto4** [comment]
It looks like Unity is also able to break visuals when we have different rotations/scales combined in:
![image](https://github.com/FlaxEngine/FlaxEngine/assets/3464853/55d7b6fd-7fd0-48b0-a293-60e02c4e38fb)

**Comment by mafiesto4** [comment]
Fixed in 9a21cfd092621fdb5bc8bfe31e6d968c8095094a.
